lib/pos_blockchain.go

```diff
@@ -1851,22 +1832,81 @@ func (bc *Blockchain) GetCommittedTipView() (*UtxoView, error) {
 }
 
+// BlockViewAndUtxoOps is a struct that contains a UtxoView and the UtxoOperations
+// and a block that were used to build the UtxoView. This struct is only
+// used for Blockchain's blockViewCache, which is used to speed up repeated access
+// to a utxo view at an uncommitted block. Simply having a utxo view was insufficient
+// for all performance enhancements as the utxo operations are needed when committing
+// a block and the block is needed for the state syncer.
+type BlockViewAndUtxoOps struct {
+	UtxoView *UtxoView
+	UtxoOps  [][]*UtxoOperation
+	Block    *MsgDeSoBlock
+}
+
+func (viewAndUtxoOps *BlockViewAndUtxoOps) Copy() (*BlockViewAndUtxoOps, error) {
+	copiedView, err := viewAndUtxoOps.UtxoView.CopyUtxoView()
+	if err != nil {
+		return nil, errors.Wrapf(err, "BlockViewAndUtxoOps.Copy: Problem copying UtxoView")
+	}
+	return &BlockViewAndUtxoOps{
+		UtxoView: copiedView,
+		UtxoOps:  viewAndUtxoOps.UtxoOps,
+		Block:    viewAndUtxoOps.Block,
+	}, nil
+}
+
 // GetUncommittedTipView builds a UtxoView to the uncommitted tip.
 func (bc *Blockchain) GetUncommittedTipView() (*UtxoView, error) {
 	// Connect the uncommitted blocks to the tip so that we can validate subsequent blocks
-	return bc.getUtxoViewAtBlockHash(*bc.BlockTip().Hash)
+	blockViewAndUtxoOps, err := bc.getUtxoViewAtBlockHash(*bc.BlockTip().Hash)
+	if err != nil {
+		return nil, errors.Wrapf(err, "GetUncommittedTipView: Problem getting UtxoView at block hash")
+	}
+	return blockViewAndUtxoOps.UtxoView, nil
 }
 
-// getUtxoViewAtBlockHash builds a UtxoView to the block provided. It does this by
-// identifying all uncommitted ancestors of this block and then connecting those blocks.
-func (bc *Blockchain) getUtxoViewAtBlockHash(blockHash BlockHash) (*UtxoView, error) {
+// getUtxoViewAtBlockHash builds a UtxoView to the block provided and returns a BlockViewAndUtxoOps
+// struct by calling getUtxoViewAtBlockHashWithTipBlock with nil as the tipBlock param.
+func (bc *Blockchain) getUtxoViewAtBlockHash(blockHash BlockHash) (*BlockViewAndUtxoOps, error) {
+	return bc.getUtxoViewAtBlockHashWithTipBlock(blockHash, nil)
+}
+
+// getUtxoViewAtBlockHashWithTipBlock builds a UtxoView to the block provided and returns a BlockViewAndUtxoOps
+// struct containing UtxoView, the UtxoOperations that resulted from connecting the block, and the full
+// block (MsgDeSoBlock) for convenience. that came from connecting the block. TipBlock is a MsgDeSoBlock of the
+// blockHash provided and should only be provided if the block is not yet stored as is the case in
+// validateAndIndexBlockPoS. It does this by identifying all uncommitted ancestors of this block. Then it checks the
+// block view cache to see if we have already computed this view. If not, connecting the uncommitted ancestor blocks
+// and saving to the cache.
+func (bc *Blockchain) getUtxoViewAtBlockHashWithTipBlock(blockHash BlockHash, tipBlock *MsgDeSoBlock) (
```

Comment:
See my comment above on reverting the changes to this function. 

Alternatively, if you want to keep the function, please simplify the signature by having the caller pass in a `*BlockNode` instead of a tip block and block hash.
```suggestion
func (bc *Blockchain) getUtxoViewAtBlockHashWithTipBlock(blockNode *BlockNode) (
```